src/amqpvalue.c

```diff
@@ -3183,22 +3960,246 @@ static int encode_array(AMQPVALUE_ENCODER_OUTPUT encoder_output, void* context,
 
         if (result == 0)
         {
+            AMQP_TYPE array_type = AMQP_TYPE_UNKNOWN;
+
             for (i = 0; i < count; i++)
             {
-                if (amqpvalue_encode(items[i], encoder_output, context) != 0)
+                AMQP_VALUE_DATA* value_data = (AMQP_VALUE_DATA*)items[i];
+                if ((array_type != AMQP_TYPE_UNKNOWN) && (value_data->type != array_type))
                 {
+                    LogError("Item type '%d' inconsistent with array type '%d'", (int)value_data->type, (int)array_type);
```

Comment:
>(int)value_data->type [](start = 81, length = 21)

Consider using ENUM_TO_STRING macro to print the nice stringified value in the enum.
uAMQP was written some time ago and I think not all of it uses it.

That would require having the enum values defined like 

#define AMQP_VALUE_TYPE_VALUES \
  AMQP_UNKNOWN, \
  ...

and then using the DEFINE_ENUM macro to actually have the enum defined and then DEFINE_ENUM_STRINGS to have a char* array with the stringified version of the values.

A consider, if it seems too much skip it please :-). Just thought of mentioning that that is how we would write it today :-).

Reply:
I've refactored out the array type check - so I will leave this for now.